src/ledger/LedgerState.cpp

```diff
@@ -1369,12 +1367,16 @@ LedgerStateRoot::Impl::getBestOffer(Asset const& buying, Asset const& selling,
     // that the lists of best offers remain properly sorted. The sort order is
     // that determined by loadBestOffers and isBetterOffer (both induce the same
     // order).
+    BestOffersCacheEntry emptyCacheEntry{{}, false};
     auto cacheKey = getBestOffersCacheKey(buying, selling);
-    if (!mBestOffersCache->exists(cacheKey))
+    if (!mBestOffersCache.exists(cacheKey))
     {
-        putInBestOffersCache(cacheKey, {{}, false});
+        putInBestOffersCache(cacheKey, emptyCacheEntry);
     }
-    auto& cached = getFromBestOffersCache(cacheKey);
+
+    // Have to check again in case the max cache size is 0 or the put throws
+    auto& cached = mBestOffersCache.exists(cacheKey)
```

Comment:
That means that `getFromBestOffersCache` can not throw anymore (as it is only used here).
What about making it `getFromBestOffersCache(key, BestOffersCacheEntry &default)`
and using as `auto& cached = getFromBestOffersCache(cacheKey, emptyCacheEntry)`